File: android/signin/app/src/main/java/com/google/samples/quickstart/signin/FBLoginActivity.java
package com.google.samples.quickstart.signin;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;

import com.facebook.AccessToken;
import com.facebook.CallbackManager;
import com.facebook.FacebookCallback;
import com.facebook.FacebookException;
import com.facebook.login.LoginManager;
import com.facebook.login.LoginResult;
import com.facebook.login.widget.LoginButton;

public class FBLoginActivity extends AppCompatActivity {
    CallbackManager callbackManager;
    LoginButton loginButton;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_fblogin);


        AccessToken accessToken = AccessToken.getCurrentAccessToken();
        boolean isLoggedIn = accessToken != null && !accessToken.isExpired();


        callbackManager = CallbackManager.Factory.create();
        LoginManager.getInstance().registerCallback(callbackManager,
                new FacebookCallback<LoginResult>() {
                    @Override
                    public void onSuccess(LoginResult loginResult) {
                        Log.e("FBLoginActivity","CallbackManageronSuccess:"+loginResult.getAccessToken().getUserId());
                        Log.e("FBLoginActivity","CallbackManageronSuccess:"+loginResult.getAccessToken().getToken());
                    }

                    @Override
                    public void onCancel() {
                        Log.e("FBLoginActivity","CallbackManageronCancel");
                    }

                    @Override
                    public void onError(FacebookException exception) {
                        Log.e("FBLoginActivity","CallbackManageronError:"+exception.getLocalizedMessage());
                    }
                });

        loginButton = (LoginButton) findViewById(R.id.login_button);
        loginButton.setReadPermissions("email");
        // Callback registration
        loginButton.registerCallback(callbackManager, new FacebookCallback<LoginResult>() {
            @Override
            public void onSuccess(LoginResult loginResult) {
                Log.e("FBLoginActivity","onSuccess:"+loginResult.getAccessToken().getUserId());
                Log.e("FBLoginActivity","onSuccess:"+loginResult.getAccessToken().getToken());
            }

            @Override
            public void onCancel() {
                Log.e("FBLoginActivity","onCancel");
            }

            @Override
            public void onError(FacebookException exception) {
                Log.e("FBLoginActivity","onError:"+exception.getLocalizedMessage());
            }
        });
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        callbackManager.onActivityResult(requestCode, resultCode, data);
        super.onActivityResult(requestCode, resultCode, data);
    }
}

File: android/signin/app/src/main/java/org/hanvic/testdemo/FBLoginActivity.kt
package org.hanvic.testdemo

import android.os.Bundle
import android.os.PersistableBundle
import android.util.Log
import android.view.ViewGroup
import android.widget.Button
import com.example.testlibrary.FBLogin
import com.example.testlibrary.FBLoginListener
import com.facebook.FacebookException
import com.facebook.login.LoginResult

class FBLoginActivity : FakeActivityX(), FBLoginListener {
    private var fb : FBLogin? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        fb = FBLogin(this)

        val btn = Button(this)
        btn.setText("FB Login")
        btn.setOnClickListener {
            fb?.login()
        }
        addContentView(btn, ViewGroup.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT))
    }

    override fun onSuccess(result: LoginResult?) {
        Log.e("FBLoginActivity", "onSuccess")
    }

    override fun onCancel() {
        Log.e("FBLoginActivity", "onCancel")
    }

    override fun onError(error: FacebookException) {
        Log.e("FBLoginActivity", "onError:${error.localizedMessage}")
    }
}